lib/primitives/mtx.js

```diff
@@ -1688,33 +1720,6 @@ class CoinSelector {
    */
 
   fund() {
-    // Ensure all preferred inputs first.
-    if (this.inputs.size > 0) {
-      const coins = [];
-
-      for (let i = 0; i < this.inputs.size; i++)
```

Comment:
Here instead of pushing blindly, we check if `view` has it, if it already has it we don't push this. Everything else can remain same.